### PR TITLE
core(audit): Ignore href=javascript:.* for rel=noopener audit

### LINKED ISSUE #3079: rel=noopener audit should ignore href=javascript:.*
discovered in #3074 

auditing google.com flags `<a href="javascript:void(0)" target="_blank">` as a failure

### PR DESCRIPTION
**Pull request by karanjthakkar**
Fixes #3079 

---

~PS: Happy to close this if @akonchady has another implementation in place already.~ https://github.com/GoogleChrome/lighthouse/issues/3079#issuecomment-336839105

### CODE REVIEW

**Comment by karanjthakkar** [comment]
@patrickhulce I'm not sure how to fix the commitlint error. It doesn't seem to consider rebased commits. Everything else seems to be passing.

File: lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
           return true;
         }
       })
+      .filter(anchor => {
+        // Ignore href's that do not redirect to a new url
+        return !/javascript:.*/.test(anchor.href);
```
Comment (wardpeet) [inline lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: wouldn't this be better?
`return !/^javascript:/.test(anchor.href);` just to make sure the string starts with javascript:

Comment (karanjthakkar) [reply lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: @wardpeet Thats a good idea. Makes it much more robust. Done!

File: lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js [hunk 2]
```diff
@@ -47,6 +47,10 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
           return true;
         }
       })
+      .filter(anchor => {
+        // Ignore href's that do not redirect to a new url
+        return !/^javascript:/.test(anchor.href);
```
[the remark this answers is not in the document]
Comment (karanjthakkar) [reply lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: Done

File: lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js [hunk 3]
```diff
@@ -47,6 +47,10 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
           return true;
         }
       })
+      .filter(anchor => {
+        // Ignore href's that do not redirect to a new url
```
[the remark this answers is not in the document]
Comment (karanjthakkar) [reply lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: Done

**Comment by karanjthakkar** [comment]
@paulirish I was going through #3490 and based on the tests (https://github.com/GoogleChrome/lighthouse/pull/3490/files#diff-8f19451b89a25fa6b1259f337761d2cbR59) I think we should be doing a case insensitive check of `javascript:`, unless i'm mistaken. Thoughts?

**Comment by karanjthakkar** [comment]
@brendankenny I meant that `startsWith` does not do a case sensitive check. So in cases where the href is `JAVASCRIPT:void(0)`, even though the browser may accept it (as mentioned in the spec linked by you), this will fail the check that I do here: https://github.com/GoogleChrome/lighthouse/pull/3574/commits/dd8e472a17b25b501ce7e9f33c6f49199f8df1e3.

**Comment by karanjthakkar** [comment]
Bump.

**Comment by wardpeet** [comment]
we could do `.toLowerCase().startsWith` just to make sure can you add a test that includes 
`'JAVASCRIPT:'` which will fail atm.

@karanjthakkar Chrome Dev Summit just happend so the team had some other priorities. Thanks for being so patient!

**Comment by karanjthakkar** [comment]
@wardpeet So sorry about that. It totally slipped my mind. I appreciate you circling back with a reply. I've made the changes 👍

File: lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js [hunk 4]
```diff
@@ -47,6 +47,14 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
           return true;
         }
       })
+      .filter(anchor => {
+        // Ignore href's that are not real links
+        return (
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: nit: we could probably simplify this to `!anchor.href || !anchor.href.toLowerCase().startsWith('javascript:')`

**Pull request merged**